driver/main.c

```diff
@@ -815,6 +847,147 @@ static int ppm_release(struct inode *inode, struct file *filp)
 	return ret;
 }
 
+static int force_tp_set(int new_tp_set)
+{
+	u32 idx;
+	u32 new_val;
+	u32 curr_val;
+	for(idx = 0; idx < TP_VAL_MAX; idx++)
+	{
+		new_val = new_tp_set & (1 << idx);
+		curr_val = g_tracepoints_attached & (1 << idx);
+		if((new_val ^ curr_val) == 0)
+		{
+			// no change needed
+			continue;
+		}
+		switch(idx)
+		{
+		case SYS_ENTER:
+			if(new_val)
+			{
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 20)
+				compat_register_trace(syscall_enter_probe, "sys_enter", tp_sys_enter);
+#else
+				register_trace_syscall_enter(syscall_enter_probe);
+#endif
+			}
+			else
+			{
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 20)
+				compat_unregister_trace(syscall_enter_probe, "sys_enter", tp_sys_enter);
+#else
+				unregister_trace_syscall_enter(syscall_enter_probe);
+#endif
```

Comment:
Not sure but probably we can create some macro/inline_functions to reuse all this code for the tracepoints, it seems almost the same in `ppm_open` `ppm_release` and here